# Exercise 1: Introduction to Data Analysis (Easy)
# Objective: Understand the basic overview and significance of data analysis.
# Task:
# Write a short essay or report on the following topics:

# What is data analysis?
Data analysis is the process of inspecting, cleansing, transforming, and modeling data with the goal of discovering useful information, drawing conclusions, and supporting decision-making. It involves various techniques and tools to analyze raw data and extract meaningful insights that can inform business strategies, scientific research, and policy-making.
# Why is data analysis important in modern contexts?
Data analysis is crucial in modern contexts for several reasons. First, it helps organizations make informed decisions by providing evidence-based insights. In a world where data is generated at an unprecedented rate, the ability to analyze and interpret this data can give businesses a competitive edge. Second, data analysis can identify trends and patterns that may not be immediately apparent, allowing for proactive measures and strategic planning. Lastly, it can improve efficiency and reduce costs by optimizing processes and identifying areas for improvement.
# List and describe three areas where data analysis is applied today.

Three areas where data analysis is applied today include:
1. **Healthcare**: Data analysis is used to improve patient outcomes, optimize treatment plans, and manage healthcare resources. By analyzing patient data, healthcare providers can identify trends in diseases, predict outbreaks, and personalize treatments based on individual patient profiles.
2. **Finance**: In the financial sector, data analysis is essential for risk management, fraud detection, and investment strategies. Financial institutions analyze market data, customer behavior, and transaction patterns to make informed decisions and mitigate risks.
3. **Marketing**: Data analysis helps businesses understand consumer behavior, segment markets, and measure the effectiveness of marketing campaigns. By analyzing customer data, companies can tailor their marketing strategies to target specific audiences and improve customer engagement.

## Exercise 2: Dataset Loading and Initial Analysis

**Objective:** Practice dataset loading from Kaggle and initial analysis.

**Task:** For the following datasets:
- How Much Sleep Do Americans Really Get?
- Global Trends in Mental Health Disorder
- Credit Card Approvals

1. Load the dataset into Jupyter or Google Colab
2. Display the first few rows
3. Provide a brief dataset description



In [ ]:
import pandas as pd

# Load the dataset
df = pd.read_csv("sleep_data.csv")

# Display first few rows
df.head()
# Basic statistics
df.describe()
# Column names & data types
df.dtypes

index                         int64
Year                          int64
Period                          str
Avg hrs per day sleeping    float64
Standard Error              float64
Type of Days                    str
Age Group                       str
Activity                        str
Sex                             str
dtype: object

# Exercise 3: Identifying Data Types


# Objective: Learn to identify different data types.
# Task:
For the datasets from the previous exercise, categorize each column of it as either quantitative or qualitative and explain your reasoning.

In [16]:
import pandas as pd

# ── Load Dataset ───────────────────────────────────────────
sleep_df = pd.read_csv("sleep_data.csv")

# ── Step 1: Quick overview ─────────────────────────────────
print("=" * 55)
print("  DATASET: How Much Sleep Do Americans Really Get?")
print("=" * 55)
print(f"\n Shape: {sleep_df.shape[0]} rows × {sleep_df.shape[1]} columns")
print(f"\n Columns & Raw dtypes:\n{sleep_df.dtypes}")

# ── Step 2: Categorize each column ─────────────────────────
column_types = {
    "index": {
        "dtype"   : sleep_df["index"].dtype,
        "type"    : "Quantitative – Discrete",
        "reason"  : "Auto-generated row ID; whole numbers, no decimals"
    },
    "Year": {
        "dtype"   : sleep_df["Year"].dtype,
        "type"    : "Quantitative – Discrete",
        "reason"  : "Represents a calendar year (2003, 2004...); countable whole numbers"
    },
    "Period": {
        "dtype"   : sleep_df["Period"].dtype,
        "type"    : "Qualitative – Nominal",
        "reason"  : f"Text label (e.g. 'Annual'); no numeric meaning. Unique: {sleep_df['Period'].unique()}"
    },
    "Avg hrs per day sleeping": {
        "dtype"   : sleep_df["Avg hrs per day sleeping"].dtype,
        "type"    : "Quantitative – Continuous",
        "reason"  : "Decimal hours (e.g. 8.57); measurable on a continuous scale"
    },
    "Standard Error": {
        "dtype"   : sleep_df["Standard Error"].dtype,
        "type"    : "Quantitative – Continuous",
        "reason"  : "Statistical margin of error; decimal values on a continuous scale"
    },
    "Type of Days": {
        "dtype"   : sleep_df["Type of Days"].dtype,
        "type"    : "Qualitative – Nominal",
        "reason"  : f"Named day categories; no natural numeric order. Unique: {sleep_df['Type of Days'].unique()}"
    },
    "Age Group": {
        "dtype"   : sleep_df["Age Group"].dtype,
        "type"    : "Qualitative – Ordinal",
        "reason"  : "Age ranges with a clear order (15–24 < 25–34 < ... < 65+)"
    },
    "Activity": {
        "dtype"   : sleep_df["Activity"].dtype,
        "type"    : "Qualitative – Nominal",
        "reason"  : f"Type of activity label; no numeric meaning. Unique: {sleep_df['Activity'].unique()}"
    },
    "Sex": {
        "dtype"   : sleep_df["Sex"].dtype,
        "type"    : "Qualitative – Nominal",
        "reason"  : f"Gender categories; no natural order. Unique: {sleep_df['Sex'].unique()}"
    },
}

# ── Step 3: Display results ────────────────────────────────
print("\n\n DATA TYPE CATEGORIZATION")
print("=" * 55)
for col, info in column_types.items():
    print(f"\n🔹 Column     : {col}")
    print(f"   Raw dtype  : {info['dtype']}")
    print(f"   Data Type  : {info['type']}")
    print(f"   Reasoning  : {info['reason']}")

# ── Step 4: Summary table ──────────────────────────────────
summary = pd.DataFrame([
    {"Column": col, "Data Type": info["type"]}
    for col, info in column_types.items()
])

print("\n\n SUMMARY TABLE")
print("=" * 55)
print(summary.to_string(index=False))

  DATASET: How Much Sleep Do Americans Really Get?

 Shape: 945 rows × 9 columns

 Columns & Raw dtypes:
index                         int64
Year                          int64
Period                          str
Avg hrs per day sleeping    float64
Standard Error              float64
Type of Days                    str
Age Group                       str
Activity                        str
Sex                             str
dtype: object


 DATA TYPE CATEGORIZATION

🔹 Column     : index
   Raw dtype  : int64
   Data Type  : Quantitative – Discrete
   Reasoning  : Auto-generated row ID; whole numbers, no decimals

🔹 Column     : Year
   Raw dtype  : int64
   Data Type  : Quantitative – Discrete
   Reasoning  : Represents a calendar year (2003, 2004...); countable whole numbers

🔹 Column     : Period
   Raw dtype  : str
   Data Type  : Qualitative – Nominal
   Reasoning  : Text label (e.g. 'Annual'); no numeric meaning. Unique: <StringArray>
['Annual']
Length: 1, dtype: str

🔹 Column   

# Exercise 4: Exploring Data Types


# Objective: Learn about different types of data in data analysis.
# Task:
Load the Iris dataset.
Identify and list which columns in your dataset are qualitative and which are quantitative.
Write a brief description of why each column is classified as qualitative or quantitative.

In [17]:
import pandas as pd

# ── Step 1: Load the Dataset ───────────────────────────────
iris_df = pd.read_csv("Iris.csv")

# ── Step 2: Display first few rows ────────────────────────
print("=" * 55)
print("         DATASET: Iris Dataset")
print("=" * 55)
print(f"\n First 5 Rows:\n")
print(iris_df.head())
print(f"\n Shape: {iris_df.shape[0]} rows × {iris_df.shape[1]} columns")
print(f"\n Raw dtypes:\n{iris_df.dtypes}")

# ── Step 3: Classify each column ──────────────────────────
column_types = {
    "Id": {
        "type"   : "Quantitative – Discrete",
        "reason" : "Whole number row identifier (1, 2, 3...); "
                   "no decimals, but not meaningful for math"
    },
    "SepalLengthCm": {
        "type"   : "Quantitative – Continuous",
        "reason" : "Measured in cm with decimals (e.g. 5.1); "
                   "can take any value on a scale"
    },
    "SepalWidthCm": {
        "type"   : "Quantitative – Continuous",
        "reason" : "Measured in cm with decimals (e.g. 3.5); "
                   "continuous physical measurement"
    },
    "PetalLengthCm": {
        "type"   : "Quantitative – Continuous",
        "reason" : "Measured in cm with decimals (e.g. 1.4); "
                   "continuous physical measurement"
    },
    "PetalWidthCm": {
        "type"   : "Quantitative – Continuous",
        "reason" : "Measured in cm with decimals (e.g. 0.2); "
                   "continuous physical measurement"
    },
    "Species": {
        "type"   : "Qualitative – Nominal",
        "reason" : "Named flower categories (Iris-setosa, "
                   "Iris-versicolor, Iris-virginica); "
                   "no numeric value or natural order"
    },
}

# ── Step 4: Print results ──────────────────────────────────
print("\n\n COLUMN CLASSIFICATION")
print("=" * 55)
for col, info in column_types.items():
    print(f"\n🔹 Column    : {col}")
    print(f"   Type      : {info['type']}")
    print(f"   Reasoning : {info['reason']}")

# ── Step 5: Summary table ──────────────────────────────────
summary = pd.DataFrame([
    {"Column": col, "Classification": info["type"]}
    for col, info in column_types.items()
])

print("\n\n SUMMARY TABLE")
print("=" * 55)
print(summary.to_string(index=False))

# ── Step 6: Grouped view ───────────────────────────────────
print("\n\n QUANTITATIVE COLUMNS:")
quant = [c for c, i in column_types.items() if "Quantitative" in i["type"]]
print("  →", quant)

print("\n  QUALITATIVE COLUMNS:")
qual = [c for c, i in column_types.items() if "Qualitative" in i["type"]]
print("  →", qual)

         DATASET: Iris Dataset

 First 5 Rows:

   Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

 Shape: 150 rows × 6 columns

 Raw dtypes:
Id                 int64
SepalLengthCm    float64
SepalWidthCm     float64
PetalLengthCm    float64
PetalWidthCm     float64
Species              str
dtype: object


 COLUMN CLASSIFICATION

🔹 Column    : Id
   Type      : Quantitative – Discrete
   Reasoning : Whole number row identifier (1, 2, 3...); no decimals, but not meaningful for math

🔹 Column    : SepalLengthCm
   Type      : Quantitative – Continuous
   Reasoning : Measured in cm wi

Quantitative Columns
These columns contain numerical values that can be measured or counted.

1. Id — Quantitative (Discrete)
This column is a unique whole-number identifier assigned to each row (1, 2, 3… 150). It is quantitative because it holds numeric values, and discrete because it only takes integer values — there is no such thing as row 2.5. However, note that while technically numeric, Id carries no meaningful mathematical value (calculating the average Id would be meaningless).

2. SepalLengthCm — Quantitative (Continuous)
This column records the length of the sepal (the outer leaf of the flower) in centimeters. It is quantitative because it is a physical measurement expressed as a number, and continuous because it can take any decimal value within a range (e.g. 4.6, 5.1, 7.9). Operations like mean, min, and max are all meaningful here.

3. SepalWidthCm — Quantitative (Continuous)
This column records the width of the sepal in centimeters. Like SepalLengthCm, it is a continuous physical measurement with decimal precision (e.g. 3.0, 3.5). It supports all arithmetic operations and is commonly used to distinguish between species.

4. PetalLengthCm — Quantitative (Continuous)
This column records the length of the petal (the inner, often colorful part of the flower) in centimeters. It is continuous and quantitative for the same reasons as the sepal measurements. Petal length is one of the most informative features in this dataset for separating species.

5. PetalWidthCm — Quantitative (Continuous)
This column records the width of the petal in centimeters. It is a continuous quantitative variable with decimal values (e.g. 0.2, 1.8). Along with petal length, it is highly useful for classifying iris species and detecting patterns.

Qualitative Column
This column contains categories or labels, not numbers.

6. Species — Qualitative (Nominal)
This column identifies which of the three iris species each flower belongs to: Iris-setosa, Iris-versicolor, or Iris-virginica. It is qualitative because it represents a label, not a measurement. It is specifically nominal (not ordinal) because there is no natural ranking or order between the three species — none is "greater than" or "less than" another. This column is typically used as the target variable in classification tasks.

# Exercise 5: Basic Observation Skills in Data Analysis


# Objective: Develop observation skills for data analysis.
# Task:
Load the How Much Sleep Do Americans Really Get? dataset into Jupyter or Google Colab.
Identify columns that could be interesting for a specific type of analysis (e.g., trend analysis, group comparison) and explain your choice.

In [9]:
import pandas as pd

# ── Step 1: Load Dataset ───────────────────────────────────
sleep_df = pd.read_csv("sleep_data.csv")

print("=" * 55)
print("   DATASET: How Much Sleep Do Americans Really Get?")
print("=" * 55)
print(f"\n First 5 Rows:\n{sleep_df.head()}")
print(f"\n Shape : {sleep_df.shape[0]} rows × {sleep_df.shape[1]} columns")
print(f"\n Year Range : {sleep_df['Year'].min()} – {sleep_df['Year'].max()}")

# ── Step 2: Identify Interesting Columns ──────────────────

# ANALYSIS 1 — Trend Analysis
print("\n\n TREND ANALYSIS: Sleep over time")
print("-" * 45)
trend = (
    sleep_df[sleep_df["Sex"] == "Both"]
    .groupby("Year")["Avg hrs per day sleeping"]
    .mean()
    .reset_index()
)
print(trend.to_string(index=False))
print("→ Columns used: Year + Avg hrs per day sleeping")

# 👥 ANALYSIS 2 — Group Comparison by Sex
print("\n\n👥 GROUP COMPARISON: Sleep by Sex")
print("-" * 45)
by_sex = (
    sleep_df.groupby("Sex")["Avg hrs per day sleeping"]
    .mean()
    .reset_index()
    .rename(columns={"Avg hrs per day sleeping": "Avg Sleep (hrs)"})
)
print(by_sex.to_string(index=False))
print("→ Columns used: Sex + Avg hrs per day sleeping")

# ANALYSIS 3 — Group Comparison by Age Group
print("\n\n GROUP COMPARISON: Sleep by Age Group")
print("-" * 45)
by_age = (
    sleep_df[sleep_df["Sex"] == "Both"]
    .groupby("Age Group")["Avg hrs per day sleeping"]
    .mean()
    .reset_index()
    .sort_values("Avg hrs per day sleeping", ascending=False)
)
print(by_age.to_string(index=False))
print("→ Columns used: Age Group + Avg hrs per day sleeping")

# ANALYSIS 4 — Weekday vs Weekend Comparison
print("\n\n COMPARISON: Weekday vs Weekend Sleep")
print("-" * 45)
by_daytype = (
    sleep_df[sleep_df["Sex"] == "Both"]
    .groupby("Type of Days")["Avg hrs per day sleeping"]
    .mean()
    .reset_index()
    .sort_values("Avg hrs per day sleeping", ascending=False)
)
print(by_daytype.to_string(index=False))
print("→ Columns used: Type of Days + Avg hrs per day sleeping")

   DATASET: How Much Sleep Do Americans Really Get?

 First 5 Rows:
   index  Year  Period  Avg hrs per day sleeping  Standard Error Type of Days  \
0      0  2003  Annual                      8.57           0.018     All days   
1      1  2004  Annual                      8.55           0.026     All days   
2      2  2005  Annual                      8.62           0.023     All days   
3      3  2006  Annual                      8.63           0.024     All days   
4      4  2007  Annual                      8.57           0.024     All days   

           Age Group  Activity   Sex  
0  15 years and over  Sleeping  Both  
1  15 years and over  Sleeping  Both  
2  15 years and over  Sleeping  Both  
3  15 years and over  Sleeping  Both  
4  15 years and over  Sleeping  Both  

 Shape : 945 rows × 9 columns

 Year Range : 2003 – 2017


 TREND ANALYSIS: Sleep over time
---------------------------------------------
 Year  Avg hrs per day sleeping
 2003                  8.702857
 2004   

# Exercise 6: Identifying Data Types
Below are various data sources. Identify whether each one is an example of structured or unstructured data.

1. A company’s financial reports stored in an Excel file.
2. Photographs uploaded to a social media platform.
3. A collection of news articles on a website.
4. Inventory data in a relational database.
5. Recorded interviews from a market research study.

Data Type Identification
1. A company’s financial reports stored in an Excel file: Structured

2. Photographs uploaded to a social media platform: Unstructured

3. A collection of news articles on a website: Unstructured

4. Inventory data in a relational database: Structured

5. Recorded interviews from a market research study: Unstructured

# Exercise 7: Transformation Exercise
For each of the following unstructured data sources, propose a method to convert it into structured data. Explain your reasoning.

A series of blog posts about travel experiences.
Audio recordings of customer service calls.
Handwritten notes from a brainstorming session.
A video tutorial on cooking.

To effectively perform analysis, we must transform unstructured data into structured formats (rows and columns) that computers can query, sort, and model. Here are proposed methods for your specified sources:

1. Blog Posts about Travel Experiences
Method: Natural Language Processing (NLP) / Named Entity Recognition (NER).

Reasoning: Blog posts contain rich, subjective text. Using Python libraries like spaCy or NLTK, you can extract specific entities such as "Location," "Budget," "Rating," and "Activities." You can also perform Sentiment Analysis to assign a score (e.g., -1 to +1) to the traveler's experience. This converts narrative text into a structured dataframe where each row is a post and columns are extracted attributes.

2. Audio Recordings of Customer Service Calls
Method: Automatic Speech Recognition (ASR) + Text Mining.

Reasoning: Audio is inherently unstructured. First, you must use a transcription service (like OpenAI's Whisper or Google Cloud Speech-to-Text) to convert audio into raw text. Once transcribed, you can use Topic Modeling (such as Latent Dirichlet Allocation) to categorize the call (e.g., "Billing Issue," "Technical Support," "Cancellation"). This produces a structured table with columns like Call_Date, Topic_Category, Sentiment, and Resolution_Status.

3. Handwritten Notes from Brainstorming
Method: Optical Character Recognition (OCR) + Semantic Tagging.

Reasoning: The primary challenge here is digitizing the physical medium. Tools like Tesseract or Google Cloud Vision can convert images of handwriting into digital text. Once digitized, you can parse the text using regex (regular expressions) to identify key components like "Idea Title," "Action Items," or "Assigned Person," transforming a chaotic jumble of thoughts into a structured project management table.

4. Video Tutorial on Cooking
Method: Computer Vision (CV) + Metadata Extraction.

Reasoning: To structure video, you must synchronize visual data with timing. You can use Frame Extraction to identify specific ingredients shown on screen (using object detection models like YOLO). By pairing these identified objects with the video's timestamp and the audio transcript (the spoken instructions), you can create a structured "Recipe Recipe-Log" containing columns like Time_Marker, Action, Ingredients_Used, and Tools_Required

# Exercise 8 : Import a file from Kaggle
Note: This dataset was originally sourced from Kaggle, but for easier access, it has been made available on GitHub.

Please download the dataset directly from the GitHub repository to use it in this project.

Import the train dataset. Use the train.csv file.
Print the first few rows of the DataFrame.

In [12]:
import pandas as pd
import requests
import zipfile
import io

# ── Step 1: Download the ZIP file from GitHub ─────────────
url = "https://github.com/devtlv/Datasets-DA-Bootcamp-2-/raw/refs/heads/main/Week%204%20-%20Data%20Understanding/W4D3%20-%20Importing%20Data%2C%20Exporting%20D/train.zip"

response = requests.get(url)

# ── Step 2: Extract the ZIP in memory ─────────────────────
with zipfile.ZipFile(io.BytesIO(response.content)) as z:
    with z.open("train.csv") as f:
        train_df = pd.read_csv(f)

# ── Step 3: Print first few rows ──────────────────────────

print(train_df.head())



   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


 # Exercise 9 : Export a dataframe to excel format and JSON format.
Create a simple dataframe.
Export the dataframe to an excel file.
Export the dataframe to a JSON file.

In [14]:
import pandas as pd

# ── Step 1: Create a Simple DataFrame ─────────────────────
data = {
    "Name"      : ["Alice", "Bob", "Charlie", "Diana", "Eve"],
    "Age"       : [25, 30, 35, 28, 22],
    "City"      : ["New York", "Paris", "London", "Tokyo", "Sydney"],
    "Salary"    : [55000, 72000, 68000, 81000, 49000],
    "Department": ["HR", "Engineering", "Marketing", "Finance", "Design"]
}

df = pd.DataFrame(data)

print("=" * 50)
print("        CREATED DATAFRAME")
print("=" * 50)
print(df.to_string(index=False))
print(f"\n Shape: {df.shape[0]} rows × {df.shape[1]} columns")

# ── Step 2: Export to Excel ────────────────────────────────
excel_file = "employees.xlsx"
df.to_excel(excel_file, index=False, sheet_name="Employees")
print(f"\n Excel file saved  → {excel_file}")

# ── Step 3: Export to JSON ─────────────────────────────────
json_file = "employees.json"
df.to_json(json_file, orient="records", indent=4)
print(f" JSON file saved   → {json_file}")

# ── Step 4: Verify — Reload both files ────────────────────
print("\n\n VERIFICATION — Reloading saved files")
print("=" * 50)

# Reload Excel
df_excel = pd.read_excel(excel_file)
print(f"\n Reloaded from Excel ({excel_file}):")
print(df_excel.to_string(index=False))

# Reload JSON
df_json = pd.read_json(json_file)
print(f"\n Reloaded from JSON ({json_file}):")
print(df_json.to_string(index=False))

        CREATED DATAFRAME
   Name  Age     City  Salary  Department
  Alice   25 New York   55000          HR
    Bob   30    Paris   72000 Engineering
Charlie   35   London   68000   Marketing
  Diana   28    Tokyo   81000     Finance
    Eve   22   Sydney   49000      Design

 Shape: 5 rows × 5 columns

 Excel file saved  → employees.xlsx
 JSON file saved   → employees.json


 VERIFICATION — Reloading saved files

 Reloaded from Excel (employees.xlsx):
   Name  Age     City  Salary  Department
  Alice   25 New York   55000          HR
    Bob   30    Paris   72000 Engineering
Charlie   35   London   68000   Marketing
  Diana   28    Tokyo   81000     Finance
    Eve   22   Sydney   49000      Design

 Reloaded from JSON (employees.json):
   Name  Age     City  Salary  Department
  Alice   25 New York   55000          HR
    Bob   30    Paris   72000 Engineering
Charlie   35   London   68000   Marketing
  Diana   28    Tokyo   81000     Finance
    Eve   22   Sydney   49000      Desig

# Exercise 10: Reading JSON Data
Use a sample JSON dataset

Import the JSON data from the provided URL.
Use Pandas to read the JSON data.
Display the first five entries of the data.

In [15]:
import pandas as pd
import requests
import zipfile
import io

# ── Method 1: Import from GitHub URL (as instructed) ──────
url = "https://github.com/devtlv/Datasets-DA-Bootcamp-2-/raw/refs/heads/main/Week%204%20-%20Data%20Understanding/W4D3%20-%20Importing%20Data%2C%20Exporting%20D/posts.zip"

response = requests.get(url)

# Extract ZIP in memory and read the JSON
with zipfile.ZipFile(io.BytesIO(response.content)) as z:
    with z.open("posts.json") as f:
        posts_df = pd.read_json(f)

# ── Display dataset info ───────────────────────────────────
print(posts_df.head())

   userId  id                                              title  \
0       1   1  sunt aut facere repellat provident occaecati e...   
1       1   2                                       qui est esse   
2       1   3  ea molestias quasi exercitationem repellat qui...   
3       1   4                               eum et est occaecati   
4       1   5                                 nesciunt quas odio   

                                                body  
0  quia et suscipit\nsuscipit recusandae consequu...  
1  est rerum tempore vitae\nsequi sint nihil repr...  
2  et iusto sed quo iure\nvoluptatem occaecati om...  
3  ullam et saepe reiciendis voluptatem adipisci\...  
4  repudiandae veniam quaerat sunt sed\nalias aut...  
